server/app.js

```diff
@@ -0,0 +1,65 @@
+const express = require('express');
+const bodyparser = require('body-parser');
+const db = require('./db/index.js');
+
+const app = express();
+
+app.set('port', 3004);
+
+app.use(bodyparser.json());
+
+app.use(bodyparser.urlencoded({extended: true}));
```

Comment:
Why is this necessary?

Reply:
Express 4 development has made Express an independent routing and middleware web framework while removing all built-in middleware from its core and hence Express versioning and releases are not affected by middleware updates.

Instead it asks user to add all the middleware as per one's requirement for ex. body-parser. 

app.use(bodyParser.json())is used to tell the system that you want json parser to be used.

bodyParser.urlencoded({extended: ...}) is used to tell the system whether you want to use a simple algorithm for shallow parsing (i.e. false) or complex algorithm for deep parsing that can deal with nested objects (i.e. true).

URL encoding and JSON encoding both allow to convert a (nested) object to string, but the format is different.

This is what I have come across from internet but you have any particular question you can specify.

Reply:
But why do you need to use `bodyParser`? Your API just accepts GET requests. It doesn't need to deal with POST requests, which is what `bodyParser` helps with.